optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -106,24 +118,46 @@ def __init__(
         self.text_encoder_model_path = Path(text_encoder_session._model_path)
         self.unet = ORTModelUnet(unet_session, self)
         self.unet_model_path = Path(unet_session._model_path)
+
+        if vae_encoder_session is not None:
+            self.vae_encoder_model_path = Path(vae_encoder_session._model_path)
+            self.vae_encoder = ORTModelVaeEncoder(vae_encoder_session, self)
+        else:
+            self.vae_encoder_model_path = None
+            self.vae_encoder = None
+
         self.tokenizer = tokenizer
         self.scheduler = scheduler
         self.feature_extractor = feature_extractor
         self.safety_checker = None
+
         sub_models = {
             DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER: self.text_encoder,
             DIFFUSION_MODEL_UNET_SUBFOLDER: self.unet,
             DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER: self.vae_decoder,
+            DIFFUSION_MODEL_VAE_ENCODER_SUBFOLDER: self.vae_encoder,
         }
+
+        # Modify config to keep the resulting model compatible with diffusers pipelines
```

Comment:
I don't understand what this means. Is it: if we use `save_pretrained` on this pipeline, the resulting model folder will be loadable in diffusers pipeline?

Reply:
Yes it allows the resulting model to be compatible with the `diffusers` pipelines (such as `OnnxStableDiffusionPipeline`) which use the model config to determine how the model should be loaded (it needs to be loaded with diffusers `OnnxRuntimeModel`).

Reply:
@patrickvonplaten, what do you think ?

Reply:
I think long-term we want to deprecate `OnnxStableDiffusionPipeline` from `diffusers` actually, but for now I think this makes sense!

Reply:
Yes will remove it once the `OnnxStableDiffusionPipeline` is deprecated